### ISSUE TITLE
TransformICCProfile() - expose input ICC profile parameter?

### ISSUE THREAD

**Issue by michaylov** [davidbyttow/govips#314]
Hello all, 
Do you think exposing the fallback input ICC profile would make sense?

Currently, the TransformICCProfile() function takes only one parameter - the output profile. The input profile is a hardcoded sRGB one.

![image](https://user-images.githubusercontent.com/34999024/195161085-1413a812-68cc-4eea-a933-1e229e7d4b3a.png)

While this is a clean solution and would cover 98% of all use cases, it does not permit to deal with some "broken" images like CMYK images without ICC profiles. And there is no control over color transformations on any images without embedded profiles.

I would expect that such an addition would also provide a solution to this issue (https://github.com/davidbyttow/govips/issues/275).

Alternatively, someting can be done for OptimizeICCProfile()? Again, nothing can do any good for CMYK images without ICC profiles.

![image](https://user-images.githubusercontent.com/34999024/195161331-89ac159c-86d6-4c85-a640-3b387472c065.png)

I would appreciate your input.

**Comment by AttilaTheFun** [comment]
@michaylov I agree it would be nice to have an input profile parameter as well. Maybe there could be a method this calls through like:
```
TransformICCProfileWithInputAndOutputProfiles(inputProfilePath, outputProfilePath string, embedded bool)
```
So existing code would call TransformICCProfile(outputProfilePath string) which would call this with the existing behavior but you could call the underlying method if needed.
